apm-lambda-extension/extension/http_server.go

```diff
@@ -18,50 +18,27 @@
 package extension
 
 import (
+	"log"
 	"net"
 	"net/http"
-	"time"
 )
 
-type serverHandler struct {
-	data   chan AgentData
-	config *extensionConfig
-}
-
-func (handler *serverHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if r.URL.Path == "/intake/v2/events" {
-		handleIntakeV2Events(handler, w, r)
-		return
-	}
-
-	if r.URL.Path == "/" {
-		handleInfoRequest(handler, w, r)
-		return
-	}
-
-	// if we have not yet returned, 404
-	w.WriteHeader(http.StatusNotFound)
-	w.Write([]byte("404"))
-
-}
+var agentDataServer *http.Server
 
-func NewHttpServer(dataChannel chan AgentData, config *extensionConfig) *http.Server {
-	var handler = serverHandler{data: dataChannel, config: config}
-	timeout := time.Duration(config.dataReceiverTimeoutSeconds) * time.Second
-	s := &http.Server{
-		Addr:           config.dataReceiverServerPort,
-		Handler:        &handler,
-		ReadTimeout:    timeout,
-		WriteTimeout:   timeout,
-		MaxHeaderBytes: 1 << 20,
-	}
+func StartHttpServer(agentDataChan chan AgentData, config *extensionConfig) (err error) {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", handleInfoRequest(config.apmServerUrl))
+	mux.HandleFunc("/intake/v2/events", handleIntakeV2Events(agentDataChan))
+	agentDataServer = &http.Server{Addr: config.dataReceiverServerPort, Handler: mux}
```

Comment:
Question: I noticed we've lost the ability to set/configure the timeouts and `MaxHeaderBytes` here.  Do you know what the defaults are and whether they're appropriate for our use case?

Reply:
I didn't mean to remove this configuration, thanks for catching that.
In looking at the `http.Server` documentation, it looks like `ReadTimeout` and friends will be set to their zero value, which is 0, if not set otherwise. The documentation says that a 0 value for `ReadTimeout` and `WriteTimeout` and means there will be no timeout.

I'd like to merge this PR so I can move forward with the explicit agent endpoint but I'll open a second PR to add this configurations back in so we can discuss there what config options are appropriate and if we want to stick with the defaults.